src/Groups/matrices/MatGrp.jl

```diff
@@ -911,19 +918,23 @@ const SU = special_unitary_group
 
 
 """
-    matrix_group(V::T...) where T<:MatrixGroup
-    matrix_group(V::AbstractVector{T}) where T<:MatrixGroup
+    matrix_group(m::Int, R::Ring, V::T...) where T<:Union{MatElem,MatrixGroupElem}
+    matrix_group(m::Int, R::Ring, V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
+    matrix_group(V::T...) where T<:Union{MatElem,MatrixGroupElem}
+    matrix_group(V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
 
-Return the matrix group generated by elements in the vector `V`.
+Return the matrix group generated by matrices in `V`. If the degree `m` and
+coefficient ring `R` are not given, then `V` must be non-empty
 """
-function matrix_group(V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
-   if T<:MatElem      # if T <: MatrixGroupElem, we can already assume that det(V[i]) != 0 
-      for v in V @assert det(v) !=0 "The matrix is not invertible" end
-   end
-   return MatrixGroup(nrows(V[1]),base_ring(V[1]),V)
+function matrix_group(m::Int, R::Ring, V::AbstractVector{T}; check::Bool=true) where T<:Union{MatElem,MatrixGroupElem}
+   return MatrixGroup(m, R, V)
 end
 
-matrix_group(V::T...) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(collect(V))
+matrix_group(m::Int, R::Ring, V::T...; check::Bool=true) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(m, R, collect(V); check)
+
+matrix_group(V::AbstractVector{T}; check::Bool=true) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(nrows(V[1]), base_ring(V[1]), V; check)
+
+matrix_group(V::T...; check::Bool=true) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(collect(V); check)
 
 matrix_group(m::Int, F::Ring) = MatrixGroup{elem_type(F), dense_matrix_type(elem_type(F))}(m, F)
```

Comment:
I introduced those `matrix_group(m::Int, F::Ring)` and `matrix_group(m::Int, F::Ring, V::AbstractVector{T})` in #2018 . They can be removed again.